localstack/services/edge.py

```diff
@@ -118,15 +118,16 @@ def forward_request(self, method, path, data, headers):
             return do_forward_request(api, method, path, data, headers, port=port)
 
     def return_response(self, method, path, data, headers, response, request_handler=None):
-
+        api, port, path, host = get_api_from_headers(headers, method=method, path=path, data=data)
```

Comment:
Thanks for this PR @duongpv7 . 👍  This may have a significant performance overhead, as we're now running `get_api_from_headers(..)` twice for each request. Happy to merge this as-is for now, but I think in a follow-up PR we should find a way to cache the value of `api`, e.g., by temporarily putting it to `headers`, as a workaround for now.